ckanext/recombinant/controller.py

```diff
@@ -182,6 +184,42 @@ def template(self, dataset_type, lang, owner_org):
             abort(404, _('Not found'))
 
         book = excel_template(dataset_type, org)
+
+        if request.method == 'POST':
+            filters = {}
+            for r in dataset['resources']:
+                if (r['name'] == dataset_type):
+                    resource = r
+
+            pk_fields = recombinant_primary_key_fields(resource['name'])
+            req_body = request.POST.get('bulk-template','')
+            raw_data = literal_eval(req_body)
+            primary_keys = []
+            chromo = get_chromo(resource['name'])
+
+            if chromo.get('edit_form'):
+                for row in raw_data:
+                    primary_keys.append(row[2:(2 + len(pk_fields))])
```

Comment:
why are you posting all the data and not just the primary keys? See the code that generates the form edit links for an example